src/components/CreatePost/CreatePost.tsx

```diff
@@ -1,50 +1,75 @@
-import React, { useContext } from 'react';
-import {
-  FormControl,
-  FormLabel,
-  FormErrorMessage,
-  Input,
-  Button,
-  Select,
-} from '@chakra-ui/react';
+import React, { useContext, useState } from 'react';
+import { FormControl, FormLabel, Input, Button } from '@chakra-ui/react';
 import { Form, Formik, Field, FieldProps, ErrorMessage } from 'formik';
 import * as Yup from 'yup';
-import RoleTag from '../RoleTag/RoleTag';
 import { FirestoreContext } from '../../services/firestore/firestore';
+import RoleList from '../RoleList/RoleList';
+import roles, { SelectedRolesInterface } from '../RoleTag/roles';
+import { StyledDescriptionInput } from './CreatePost.styles';
 
 interface FormValues {
   title: string;
   description: string;
-  roles: string[];
 }
 
-const ProjectForm = () => {
+const defaultSelectedRoles = Object.values(roles).reduce(
+  (acc, { id }) => ({
+    ...acc,
+    [id]: false,
+  }),
+  {},
+);
+
+const CreatePost = () => {
+  const [selectedRoles, setSelectedRoles] = useState<SelectedRolesInterface>(
+    defaultSelectedRoles,
+  );
   const initialValues: FormValues = {
     title: '',
     description: '',
-    roles: [],
   };
+
+  const validationSchema = Yup.object({
```

Comment:
Since this object does not use any props/state variables, this should be declared above the function.